ee/agent/timemachine/timemachine_darwin.go

```diff
@@ -0,0 +1,36 @@
+//go:build darwin
+// +build darwin
+
+package timemachine
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/kolide/launcher/ee/agent/types"
+	"github.com/kolide/launcher/ee/allowedcmd"
+)
+
+// ExcludeLauncherDB adds the launcher db to the time machine exclusions for
+// darwin and is noop for other oses
+func ExcludeLauncherDB(ctx context.Context, k types.Knapsack) {
+	dbPath := k.BboltDB().Path()
+	cmd, err := allowedcmd.Tmutil(ctx, "addexclusion", dbPath)
```

Comment:
I think we should exclude the whole root directory. Maybe the secret too?

Reply:
Actually keep the secret. But probably ignore the whole root directory

Reply:
I thought about that, but if I understand correctly, that would mean we would also not bring along all the updates. So after a time machine restore / backup the user would be back on the very first version of launcher they downloaded and would need to update again. Maybe that's okay?

Reply:
Grabbing the updates, and the TUF db, is a great counterpoint. I think you're right, it's good to carry those. 

Maybe we should drop some more stuff though. Can we go by wildcard? Or do we need to do own globbing?

Reply:
we can wild card, you thinking `*.db`?

Reply:
```
$ ls /var/kolide-k2/k2device-preprod.kolide.com/
augeas-lenses				menu.json
debug-2023-12-31T20-27-14.599.json.gz	menu_template.json
debug-2024-01-03T05-30-26.256.json.gz	metadata.json
debug-2024-01-06T19-20-54.029.json.gz	metadata.plist
debug-2024-01-09T01-44-56.485.json.gz	osquery.autoload
debug-2024-01-10T20-30-07.640.json.gz	osquery.db
debug.json				osquery.pid
desktop_501				osquery.sock
kolide.png				osquery.sock.34719
kv.sqlite				osquery.sock.3513
launcher-staging			osqueryd-staging
launcher-tuf				osqueryd-tuf
launcher-tuf-dev			osqueryd-tuf-dev
launcher-version-1.4.1-3-ge21cebc	tuf
launcher.db				tuf-dev
launcher.pid				updates
```

json, json.gz, the metadata, the menu, most of the osquery files, desktop, the pid files, sockets...

Reply:
wildcards did not work as expected, ended up globbing